test/onnx/onnx_import_provenance.in.cpp

```diff
@@ -0,0 +1,93 @@
+//*****************************************************************************
+// Copyright 2017-2020 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//*****************************************************************************
+
+#include "gtest/gtest.h"
+#include "ngraph/file_util.hpp"
+#include "ngraph/frontend/onnx_import/default_opset.hpp"
+#include "ngraph/frontend/onnx_import/onnx.hpp"
+#include "util/test_control.hpp"
+#include "util/type_prop.hpp"
+
+using namespace ngraph;
+using namespace ngraph::onnx_import;
+
+static std::string s_manifest = "${MANIFEST}";
+
+NGRAPH_TEST(onnx_${BACKEND_NAME}, provenance_tag_text)
+{
+    const auto function = onnx_import::import_onnx_model(
+        file_util::path_join(SERIALIZED_ZOO, "onnx/provenance_tag_add.prototxt"));
+
+    const auto ng_nodes = function->get_ordered_ops();
+    for (const auto ng_node : ng_nodes)
+    {
+        for (const auto tag : ng_node->get_provenance_tags())
+        {
+            EXPECT_HAS_SUBSTRING(tag, "ONNX");
+        }
+    }
+}
+
+template <typename NodeToCheck>
```

Comment:
Do you need this to be a templated function? `get_provenance_tags` is a function of the base `Node` class.

Reply:
Yes, this has to be a template because its parameter is used to find a node in the subgraph returned by `function->get_ordered_ops()`. Depending on the test I'm looking for either an op, constant or parameter and they are expected to have different provenance tags. 
This function is just a helper that reduces code duplication in unit tests.